tools/pepsi/src/upload.rs

```diff
@@ -38,6 +38,40 @@ pub struct Arguments {
     /// The NAOs to upload to e.g. 20w or 10.1.24.22
     #[arg(required = true)]
     pub naos: Vec<NaoAddress>,
+    #[arg(long)]
+    pub skip_check_os: bool,
+}
+
+pub async fn check_os(nao: &Nao) -> bool {
+    let os_stable = "5.1.1";
+    let output = nao
+        .ssh_to_nao()
+        .arg("cat /etc/os-release")
+        .output()
+        .await
+        .unwrap();
+
+    let request = String::from_utf8(output.stdout).unwrap();
+    let lines = request.lines();
+    for line in lines {
+        if line.contains("VERSION_ID") {
+            let line_split: Vec<&str> = line.split('=').collect();
+            let os_version = line_split[1];
+
+            println!("Unstable version on {}", nao.host);
```

Comment:
Could you just print `nao` instead of `nao.host`? Maybe have a look around how the other commands and the `nao` crate are printing NAOs.

Reply:
Ok

Reply:
It didn't work because you didn't change anything?